src/base.jl

```diff
@@ -667,32 +667,45 @@ end
 
 struct ServiceContributingDevices
     service::Service
-    contributing_devices::Vector{Device}
+    contributing_devices::Vector{Component}
 end
 
 const ServiceContributingDevicesKey = NamedTuple{(:type, :name), Tuple{DataType, String}}
 const ServiceContributingDevicesMapping =
     Dict{ServiceContributingDevicesKey, ServiceContributingDevices}
 
+"""
+Return a vector of devices contributing to the service.
+"""
+function _get_contributing_devices(sys::System, service::T) where {T <: Service}
+    uuid = IS.get_uuid(service)
+    devices = ServiceContributingDevices(service, Vector{Device}())
+    for device in get_components(Device, sys)
+        for _service in get_services(device)
+            if IS.get_uuid(_service) == uuid
+                push!(devices.contributing_devices, device)
+                break
+            end
+        end
+    end
+    return devices
+end
+
+"""
+Return a vector of services contributing to the group service.
+"""
+function _get_contributing_devices(sys::System, service::T) where {T <: StaticReserveGroup}
+    return get_contributing_devices(service)
+end
+
 """
 Return an instance of ServiceContributingDevicesMapping.
 """
 function get_contributing_device_mapping(sys::System)
     services = ServiceContributingDevicesMapping()
     for service in get_components(Service, sys)
-        uuid = IS.get_uuid(service)
-        devices = ServiceContributingDevices(service, Vector{Device}())
-        for device in get_components(Device, sys)
-            for _service in get_services(device)
-                if IS.get_uuid(_service) == uuid
-                    push!(devices.contributing_devices, device)
-                    break
-                end
-            end
-
-            key = ServiceContributingDevicesKey((typeof(service), get_name(service)))
-            services[key] = devices
-        end
+        key = ServiceContributingDevicesKey((typeof(service), get_name(service)))
+        services[key] = _get_contributing_devices(sys, service)
```

Comment:
It seems like this won't work if the service is a StaticReserveGroup.  The first _get_contributing_devices method returns a ServiceContributingDevices whereas the second method return a Vector.  This dictionary requires the former.

Reply:
Yes, sorry about that! there was a bug on `get_contributing_devices`.

Reply:
I will do some thinking tonight about the way I went with those functions and answer you in the morning.

Reply:
Now it should be fixed. However, since, on the new release, you guys are calling 'PSY.get_contributing_device_mapping(sys)' inside the 'construct_service' function, all this change to 'ServiceContributingDevices' is no longer needed haha I can just call 'get_contributing_services'. 
The whole thing was just to not change the 'construct_services' API.

Reply:
@daniel-thom and @jd-lara I will refactor the PR accordingly tomorrow morning to remove all the useless stuff.

Reply:
done